sechub-sereco/src/test/java/com/mercedesbenz/sechub/sereco/importer/SensitiveDataMaskingServiceTest.java

```diff
@@ -0,0 +1,206 @@
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(sechubConfigWithoutWebScan, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+    }
+
+    @Test
+    void vulnerabilities_not_changed_when_no_headers_are_configured() throws IOException {
+        /* prepare */
+        SecHubConfigurationModel scanConfigWithoutHeaders = new SecHubConfigurationModel();
+        scanConfigWithoutHeaders.setWebScan(new SecHubWebScanConfiguration());
+
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_REPORT_FILE_WITH_SENSITIVE_HEADERS);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(scanConfigWithoutHeaders, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+    }
+
+    @Test
+    void empty_sarif_report_with_no_findings_handled_without_changing_anything() throws IOException {
+        /* prepare */
+        String sechubConfigJson = TestFileReader.loadTextFile(SECHUB_WEBSCAN_CONFIG_FILE_WITH_SENSITIVE_HEADERS);
+        SecHubConfigurationModel scanConfig = SecHubConfiguration.createFromJSON(sechubConfigJson);
+
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_EMPTY_REPORT_FILE);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(scanConfig, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = ScanType.class, names = { "WEB_SCAN" }, mode = EnumSource.Mode.EXCLUDE)
+    void wrong_scan_type_results_in_unchanged_vulnerabilities(ScanType scanType) {
+        /* prepare */
+        String sechubConfigJson = TestFileReader.loadTextFile(SECHUB_WEBSCAN_CONFIG_FILE_WITH_SENSITIVE_HEADERS);
+        SecHubConfigurationModel scanConfigWithSensitiveHeaders = SecHubConfiguration.createFromJSON(sechubConfigJson);
+
+        SerecoVulnerability vulnerability = mock(SerecoVulnerability.class);
+        vulnerability.setWeb(new SerecoWeb());
+        SerecoMetaData data = new SerecoMetaData();
+        data.getVulnerabilities().add(vulnerability);
+
+        when(vulnerability.getScanType()).thenReturn(scanType);
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(scanConfigWithSensitiveHeaders, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+        verify(vulnerability, times(1)).getScanType();
+        verify(vulnerability, never()).getWeb();
+    }
+
```

Comment:
Please add tests for the illegal arguments exceptions for the null arguments for the service.